src/plugins/data/public/plugin.ts

```diff
@@ -182,6 +183,7 @@ export class DataPublicPlugin implements Plugin<DataPublicPluginSetup, DataPubli
       ui: {
         IndexPatternSelect: createIndexPatternSelect(core.savedObjects.client),
         SearchBar,
+        ensureDefaultIndexPattern: createEnsureDefaultIndexPattern(core, indexPatterns),
```

Comment:
I think this should not be exported from here.
It's not really a UI component, and the naming makes it extra unclear.
It looks more like it should be a `useDefaultIndexPattern` hook, that is exported from `kibana_react`.

Reply:
This would be the best way to implement this (except the thing about showing the warning banner, it would be disappeared right after a plugin is destroyed),
but unfortunately, right now it's only used across angular driven plugins and I guess there is no way to integrate this pattern well there

Reply:
@lizozom What do you think about exporting this from `IndexPatternsService`?

Reply:
> It looks more like it should be a useDefaultIndexPattern hook, that is exported from kibana_react

I agree the ergonomics of this would be better. The only problem is that the way this helper is currently written (with `bannerId` and `timeoutId` being stateful), means we really have no choice but to put it somewhere in the runtime contract of the `data` plugin, as `kibana_utils` and `kibana_react` are only imported statically.

I don't have strong feelings on whether it goes in `ui` or `IndexPatternsService`, but since it contains state that needs to be shared across apps, it seems those are our only options outside of refactoring the function altogether and storing that state elsewhere (e.g. sessionStorage).

Reply:
So what would be the final decision? 
Should it stay in `ui` or moved into `IndexPatternsService`, or even postponed in respect of https://github.com/elastic/kibana/issues/57401 ?

Reply:
If @lizozom doesn't like the idea of putting it in `ui`, then IMHO it's fine to put it on `IndexPatternsService`. She should be available tomorrow and can hopefully chime in if she has any other ideas.

> or even postponed in respect of #57401 ?

I don't think we should wait on that issue to fix this bug. That discussion is still in early stages & I don't know when the actual implementation will be prioritized.

Reply:
@sulemanof 

I think moving it to the index pattern service is better.
It's not an actual UI component.

Thanks!